gateway/src/types.rs

```diff
@@ -100,6 +108,18 @@ impl PostBoxItem {
     }
 }
 
+pub(crate) fn ensure_message_sorted<E: ExecutableMessage>(messages: &[E]) -> anyhow::Result<()> {
+    // check top down msgs
+    for i in 1..messages.len() {
+        match messages[i - 1].nonce().cmp(&messages[i].nonce()) {
+            Ordering::Less => {}
```

Comment:
@cryptoAtwill, I don´t think we should throw an error here. Let's just order them in case they are not ordered and return them. If the checkpoint was committed here it was because all the validators agreed on that, so we can trust the messages (is not the same as for cron checkpoints). Let's discuss if this is not clear.